src/lib/overloads.jl

```diff
@@ -162,15 +162,31 @@ Returns a subset of the argument layer, where the new limits are given by
 `left`, `right`, `top`, and `bottom`. Up to three of these can be omitted, and
 if so these limits will not be affected.
 """
-function Base.getindex(p::T; left::K=nothing, right::K=nothing, top::K=nothing, bottom::K=nothing) where {T <: SimpleSDMLayer, K <: Union{Nothing,AbstractFloat}}
-   imax = isnothing(right) ? p.right : match_longitude(p, right)
-   imin = isnothing(left) ? p.left : match_longitude(p, left)
-   jmax = isnothing(top) ? p.top : match_latitude(p, top)
-   jmin = isnothing(bottom) ? p.bottom : match_latitude(p, bottom)
+function Base.getindex(p::T; left::Union{N,A}=nothing, right::Union{N,A}=nothing, top::Union{N,A}=nothing, bottom::Union{N,A}=nothing) where {T <: SimpleSDMLayer, N <: Nothing, A <: AbstractFloat}
```

Comment:
Is there a better way to make the types work? They have to be "independent", so that one can be `Nothing` while the others `AbstractFloat`. The previous code worked only if they were all `Nothing` or all `AbstractFloat`.

Reply:
Well, we can always *not* be specific about the types here, since there is an `isnothing` on one side, and the `getindex` measures do not have methods for anything other than a float in this case... would it fix the tests?

Reply:
Oh, it already works like this. I was just wondering if there was a cleaner way to call the types.
It works the same if we're not specific about them, as `match_longitude` is specific for floats.